pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/name/InputFileSegmentNameGenerator.java

```diff
@@ -64,23 +71,25 @@ public InputFileSegmentNameGenerator(String filePathPattern, String segmentNameT
       throw new IllegalArgumentException(
           String.format("Invalid inputFileUri: %s for InputFileSegmentNameGenerator", inputFileUri), e);
     }
+    _appendUUIDToSegmentName = appendUUIDToSegmentName;
     _segmentName = makeSegmentName();
   }
 
   private String makeSegmentName() {
     String inputFilePath = _inputFileUri.getPath();
     Matcher m = _filePathPattern.matcher(inputFilePath);
+    String segmentName;
+
     if (!m.matches()) {
       LOGGER.warn(String.format("No match for pattern '%s' in '%s'", _filePathPattern, inputFilePath));
-      return safeConvertPathToFilename(inputFilePath);
+      segmentName = safeConvertPathToFilename(inputFilePath);
+    } else {
+      segmentName = _segmentNameTemplate;
+      for (int i = 1; i <= m.groupCount(); i++) {
+        segmentName = segmentName.replace(String.format(PARAMETER_TEMPLATE, i), m.group(i));
+      }
     }
-
-    String segmentName = _segmentNameTemplate;
-    for (int i = 1; i <= m.groupCount(); i++) {
-      segmentName = segmentName.replace(String.format(PARAMETER_TEMPLATE, i), m.group(i));
-    }
-
-    return segmentName;
+    return JOINER.join(segmentName, _appendUUIDToSegmentName ? UUID.randomUUID() : null);
```

Comment:
(minor) avoid unnecessary join
```suggestion
    return _appendUUIDToSegmentName ? JOINER.join(segmentName, UUID.randomUUID()) : segmentName;
```